dev/tasks/vcpkg-tests/github.windows.yml

```diff
@@ -36,6 +36,7 @@ jobs:
           git -C arrow checkout FETCH_HEAD
           git -C arrow submodule update --init --recursive
       - name: Remove and Reinstall vcpkg
+        # TODO: this comment seems outdated, the vcpkg on GHA is quite recent.
```

Comment:
Note @kou @kszucs : I tried to remove the vcpkg-reinstall step, but then got an error:
```
Error: Couldn't find explicitly specified baseline `"a267ab118c09f56f3dae96c9a4b3410820ad2f0b"` in the baseline file, and there was no baseline at that commit or the commit didn't exist.
Error: while checking out baseline 'a267ab118c09f56f3dae96c9a4b3410820ad2f0b:versions/baseline.json':
fatal: path 'versions/baseline.json' exists on disk, but not in 'a267ab118c09f56f3dae96c9a4b3410820ad2f0b'
```
https://github.com/ursacomputing/crossbow/runs/2933445520?check_suite_focus=true

Reply:
@pitrou versions of vcpkg rapidly stop working until updated, and the most reliable way to update vcpkg is simply to remove and reinstall it, as is done here. I'm not surprised that removing the reinstall step would trigger an error.

The comment lines 39–45 could be removed. I think the final two lines of this comment (`# When running...`) are probably sufficient.